### ISSUE TITLE
Performance: uniqueChars in exportToSVG includes all chars regardless of font faces

### ISSUE THREAD

**Issue by zsviczian** [excalidraw/excalidraw#8678]
While debugging #8677 I noticed that the logic in exportToSvg will result in unnecessary glyphs being included in the SVG file depending on the font faces and the text in the scene. This is because uniqueChars is not calculated by font face but for the whole batch.

@Mrazator Am I missing something?

**Comment by zsviczian** [comment]
This will actually cause problems in ExcalidrawFontFace.toCSS

when fonts will be rejected due to characters outside their range... yet those characters were intended for a different font. I can imagine this when mixed latin and cjk text is displayed.

```ts
  public toCSS(
    characters: string,
    codePoints: Array<number>,
  ): Promise<string> | undefined {
    // quick exit in case the characters are not within this font face's unicode range
    if (!this.getUnicodeRangeRegex().test(characters)) {
      return;
    }
```

this is because codePoints here represent codepoints from all text elements, not text elements with this font.

**Comment by Mrazator** [comment]
> This is because uniqueChars is not calculated by font face but for the whole batch.

Yeah, this was initially on purpose to keep things simple, as we aimed at having a common CJK fallback for any font. Hence including one more "alphabetic" font face when you already had 250 font faces didn't make much difference. Now this further optimization for alphabetic fonts might be actually quite trivial, let me check what we can do.

> when fonts will be rejected due to characters outside their range.

Could this happen really? In the worst case, more font faces will be subsetted and included in the SVG, as we have more codepoints to check - but none should really be rejected. In other words, it's enough that just one character matches the regex, it does not matter that there are also characters out of the given range.

**Comment by Mrazator** [comment]
Done in https://github.com/excalidraw/excalidraw/pull/8641
